### ISSUE TITLE
Product Flat indexer fails

### ISSUE THREAD

**Issue by zamoroka** [magento/magento2#23809]
### Preconditions
1. Magento version 2.2.9

### Steps to reproduce
1. Create a product attribute with code 'range'
2. Run `catalog_product_flat` reindex

### Expected result
1. Product flat indexer works properly 

### Actual result
1. Product flat indexer fails with error: 
```
SQLSTATE[42000]: Syntax error or access violation: 1064 You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near 'range IS NOT NULL)' at line 5, query was: UPDATE `catalog_product_flat_11_tmp_indexer` AS `et`
 LEFT JOIN `eav_attribute_option_value` AS `t0` ON t0.option_id = et.range AND t0.store_id = 0
 LEFT JOIN `eav_attribute_option_value` AS `ts` ON ts.option_id = et.range AND ts.store_id = 11
SET `et`.`range_value` = IFNULL(ts.value, t0.value)
WHERE (range IS NOT NULL)
```

**Comment by zamoroka** [comment]
@magento give me 2.2 instance

**Comment by engcom-Bravo** [comment]
Hello @zamoroka. Thanks for reporting. Currently we are not able to reproduce Your issue following the steps You provided. Would You please add some more details on how we can reproduce it?

**Comment by zamoroka** [comment]
My steps to reproduce:
1. Install clean Magento using branch `2.2`
2. Create new product attribute "Range" (Dropdown)
![image](https://user-images.githubusercontent.com/9164112/61632823-6d47ba80-ac96-11e9-8b70-9315c9331a81.png)
3. Add the attribute to "default" attribute set
4. Create a new product and set attribute value
5. Enable flat catalog for products
6. Run product flat indexer

**Comment by zamoroka** [comment]
Closing as issue is fixed in 2.3-develop
